python/yugabyte_db_thirdparty/builder_interface.py

```diff
@@ -175,3 +175,6 @@ def copy_include_files(
             rel_src_include_path: str,
             dest_include_path: str) -> None:
         raise NotImplementedError()
+
+    def prepare_for_build_tool_invocation(self, dep: 'Dependency') -> bool:
```

Comment:
This is an already implemented function and it does the following things

1. There is a skip_build_invocations command line option. And if this is set, then we should return immediately without building.
2. It also checks that we are in the correct directory for building the library.

I simply added an interface here since no other library uses the rust compiler. I need to invoke the rust compiler myself because there is no API to do so. But I need to call this function similar to what C++ builds do.